feat(integrations): add the falco integrations from the main falco repo

maxgio92 opened the pull request:
start to move the non-official and community maintained integrations of falco from the main
repository.

maxgio92 commented:
/kind feature

maxgio92 commented:
/cc @leodido

leogr commented:
LGTM!

leodido commented:
/check-dco

leodido commented on integrations/anchore-falco/README.md:
````diff
@@ -0,0 +1,89 @@
+### Build the image
+
+```
+docker build -t sysdig/anchore-falco .
````
Can we please fix such occurrences? :)

maxgio92 replied:
Do you mean to move the Docker images to another community repository?

maxgio92 replied:
Do you mean to move this repo under falcosecurity as well as for the event generator?

leogr replied:
Since there are no other references to `sysdig/anchore-falco` besides these in the README, I would suggest to just rewrote the instruction in order to use a local image (eg. `anchore-falco`).

IMHO, it would be enough. If it was actively maintained and required a new repo, we will create it later.

leodido commented on integrations/anchore-falco/README.md:
````diff
@@ -0,0 +1,89 @@
+
+### Running the image
+
+An image exists on DockerHub, its name is ```sysdig/anchore-falco```.
````
Also here please :)

maxgio92 replied:
The same as above

leodido commented on integrations/anchore-falco/README.md:
```diff
@@ -0,0 +1,89 @@
+docker run --rm -e ANCHORE_CLI_USER=<user-for-custom-anchore-engine> \
+                -e ANCHORE_CLI_PASS=<password-for-user-for-custom-anchore-engine> \
+                -e ANCHORE_CLI_URL=http://<custom-anchore-engine-host>:8228/v1 \
+                sysdig/anchore-falco
```
And here!

maxgio92 replied:
The same as above

leodido commented on integrations/k8s-using-daemonset/falco-event-generator-deployment.yaml:
```diff
@@ -0,0 +1,17 @@
+    spec:
+      containers:
+      - name: falco-event-generator
+        image: sysdig/falco-event-generator:latest
```
This needs probably to be:

1. renamed
2. coordinate with the new event-generator at https://github.com/falcosecurity/event-generator

maxgio92 replied:
I think that it can be considered done :) But I added a consideration in the last comment

leogr replied:
Atm, `falcosecurity/event-generator` is not yet ready, although a test image already exists.

@leodido WDYT ? 
We may keep this on hold until dependant issues are solved. Regarding the `event-generator`, I guess it will take another week or so.

leogr replied:
Done :point_right: https://github.com/falcosecurity/event-generator/releases/tag/v0.1.0

maxgio92 replied:
@leogr we'll keep the sysdig one for now? Or we can wait for a release of the Docker image v0.1.0? Thank you!

leogr replied:
You are welcome! 
It's released, see my review below

leodido commented on integrations/k8s-using-daemonset/k8s-with-rbac/falco-daemonset-configmap-slim.yaml:
```diff
@@ -0,0 +1,100 @@
+          #  - name: FALCOCTL_FALCO_PROBE_URL
+          #    value:
+          #  - name: FALCOCTL_FALCO_PROBE_REPO
+          #    value: "https://s3.amazonaws.com/download.draios.com/stable/sysdig-probe-binaries/"
```
This URL can be left as is for now (but it'll need to be updated soon :))

leogr replied:
I'm not sure those are working, I propose to fix this later in another PR and unlock this now since we have everything else solved (once suggestions have been applied).
@leodido @maxgio92 WDYT?

maxgio92 replied:
I agree @leogr

poiana commented:
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **maxgio92**
You can assign the PR to them by writing `/assign @maxgio92` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=falcosecurity%2Fcontrib).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/falcosecurity/contrib/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":["maxgio92"]} -->

poiana commented:
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **leodido**
You can assign the PR to them by writing `/assign @leodido` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=falcosecurity%2Fcontrib).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/falcosecurity/contrib/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":["leodido"]} -->

maxgio92 commented:
Sure @leodido ! I just replied to the specific reviews for clarifications and updated this branch because it was out-of-sync with the Falco repo's master branch.
Thank you

maxgio92 commented:
I just replaced the event generator docker repository; the anchore-falco one it seems that it doesn't  exist yet in the falcosecurity's docker hub organization.

poiana commented:
Thanks for your pull request. Before we can look at it, you'll need to add a 'DCO signoff' to your commits.

:memo: **Please follow instructions in the [contributing guide](https://github.com/falcosecurity/contrib/blob/master/CONTRIBUTING.md) to update your commits with the DCO**

Full details of the Developer Certificate of Origin can be found at [developercertificate.org](https://developercertificate.org/).

**The list of commits missing DCO signoff**:

- [fed9743](https://github.com/falcosecurity/contrib/commits/fed97437d70e981d60689b8adaaeecfcbfdf3074) improvement(event-generator): update the event-generator deployment modes

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

maxgio92 commented:
I also updated the event-generator deployment modes (k8s deployment and job) as for in the [related repository](github.com/falcosecurity/event-generator). Anyway, in my opinion it doesn't make a lot of sense, since is duplicated code and documentation (and tripled considering the falco-k8s-daemonset vs falco-k8s-deployment).
Therefore in this repo I would simply refer to the [event-generator official repository](github.com/falcosecurity/event-generator) to know how to generate sample events, without providing neither code nor guides, in order to have only one source of truth.
What do you think?

leogr commented:
> Therefore in this repo I would simply refer to the [event-generator official repository](github.com/falcosecurity/event-generator) to know how to generate sample events, without providing neither code nor guides, in order to have only one source of truth.
> What do you think?
Personally, I totally agree. Maybe the old one could just be archived in some way?

leogr commented on integrations/k8s-using-daemonset/falco-event-generator-deployment.yaml:
```diff
@@ -3,15 +3,19 @@ kind: Deployment
       - name: falco-event-generator
-        image: falcosecurity/event-generator:latest
+        image: falcosecurity/event-generator
+        command: ["./event-generator", "run"]
```
```suggestion
        command: ["./event-generator", "run", "syscall", "--loop"]
```
To retain the previous behavior we need 
 - `syscall` (otherwise also the `k8saudit` events collection will run, and probably it will not work because the k8s audit needs to be configured)
 - `--loop`  (as soon as https://github.com/falcosecurity/event-generator/pull/12 gets merged).

leogr commented on integrations/k8s-using-daemonset/falco-event-generator-job.yaml:
```diff
@@ -0,0 +1,13 @@
+      containers:
+      - name: event-generator
+        image: falcosecurity/event-generator
+        command: ["./event-generator", "run"]
```
```suggestion
        command: ["./event-generator", "run", "syscall"]
```
Same as above, but without `--loop` since here we want to generate sample events once.

leogr commented on integrations/k8s-using-deployment/falco-event-generator-deployment.yaml:
```diff
@@ -3,15 +3,19 @@ kind: Deployment
       - name: falco-event-generator
-        image: falcosecurity/event-generator:latest
+        image: falcosecurity/event-generator
+        command: ["./event-generator", "run"]
```
```suggestion
        command: ["./event-generator", "run", "syscall", "--loop"]
```
Same as above

leogr commented on integrations/k8s-using-deployment/falco-event-generator-job.yaml:
```diff
@@ -0,0 +1,13 @@
+      containers:
+      - name: event-generator
+        image: falcosecurity/event-generator
+        command: ["./event-generator", "run"]
```
```suggestion
        command: ["./event-generator", "run", "syscall"]
```
Same as above without `--loop`

leogr commented:
Hey @maxgio92 

> Therefore in this repo I would simply refer to the [event-generator official repository](github.com/falcosecurity/event-generator) to know how to generate sample events, without providing neither code nor guides, in order to have only one source of truth.
> What do you think?

I agree (see comments within my review). So, please do it! :smiley:

maxgio92 commented:
> Just minor fixes to the event-generator stuff.
> 
> Btw, I think we could move the event-generator related files to the [new event-generator repo](https://github.com/falcosecurity/event-generator/tree/master/deployment) and just link them in the documentation here. That would help with the maintenance.
> 
> Please, feel free to open a new PR in the new event-generator repo and clean them up from here.
> 
> Thanks again!

Hi @leogr, I implemented the suggestions and then cleaned the event-generator sample configurations for consistency; updated the documentation to refer to the new official event-generator's repo.

poiana commented:
LGTM label has been added.  <details>Git tree hash: 6c0e736ff8c61472b935c1592f08ba12148ea2c1</details>

poiana commented:
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/falcosecurity/contrib/pull/3#pullrequestreview-400501445" title="Approved">leogr</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=falcosecurity%2Fcontrib).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/falcosecurity/contrib/blob/master/OWNERS)~~ [leogr]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":[]} -->

poiana commented:
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/falcosecurity/contrib/pull/3#pullrequestreview-400796019" title="Approved">leodido</a>*, *<a href="https://github.com/falcosecurity/contrib/pull/3#pullrequestreview-400501445" title="Approved">leogr</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=falcosecurity%2Fcontrib).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/falcosecurity/contrib/blob/master/OWNERS)~~ [leodido,leogr]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":[]} -->
